### ISSUE TITLE
Incorrect values for preprocessor variables WINVER and _WIN32_WINNT

### ISSUE THREAD

**Issue by DavidOgilvie** [open-watcom/open-watcom-v2#793]
I am compiling a GUI C project on a Windows 10 system running 64 bit version 21H2.

The preprocessor values for WINVER and _WIN32_WINNT are both 0x0400.

Because of this, various portions of WINUSER.H are not being processed, and a lot of the window message definitions for window 7, 8 and 10 are not defined.

Should WINUSER and _WIN32_WINNT have the values appropriate to the system compiling the code, or should I be using a command-line directive to set the value for WINUSER and _WIN32_WINNT.

Any suggestions would he helpful!

Thanks in advance!

David

**Comment by ArmstrongJ** [comment]
I think you would need to manually set them to whatever value you wish based on your target system compatibility goals.  They shouldn't be set based on the compiler's host OS version because you would then be unable to produce an executable for, let's say, Windows XP on a Windows 10 system.  I use Open Watcom to build a non-trivial Windows GUI project, and I set them manually.  The build server is Windows Server something-or-other, but the application runs on Windows 2000 through Windows 11 just fine (and looks alright too).

**Comment by jmalak** [comment]
This is correct behavior, because we don't have SDK for each Version of Windows (something like Microsoft).
We have one version which cover all version from Windows NT 4.0 (it is minimal version of Windows supported).
If you want different version then it is your business to setup required macros WINVER and _WIN32_WINNT.
Take into account that C run-time library support Windows from version Windows NT 3.0 or Win32s. It is configured dynamically based on detected run-time version of Windows.

**Comment by jmalak** [comment]
Another important think is required version of Windows WIN32 subsystem. 
OW has predefined version 4.0 (Windows NT 4.0).
If you need diferent value then you must specify it by linker directive 
`runtime windows=major.minor`
